File: Projects/10/Ruby/Kiwi.rb
# Kiwi.rb | Defines the Kiwi class which inherits attributes and methods
#   from the Walkingbird superclass.
#
# Completed by: Emily Costa
# Date: 4/27/2020
####################################################

require './Walkingbird.rb'

class Kiwi < Walkingbird

    def initialize(name)
        @name = name
    end

    def className
        "Ostrich"
    end

    def call
        "Whoo-hoo!"
    end

end

File: Projects/10/Java/Kiwi.java
/* Kiwi.java provides a Kiwi subclass of Flyingbird.
 *
 * Completed by: Emily COsta
 * Date: 4/27/2020
 ******************************************************/

public class Kiwi extends Walkingbird{
    
    /* explicit constructor
     * Receive: name, a String 
     * PostCond: myName == name.
     */
    public Kiwi(String name) {
        super(name);
    }

    /* A Kiwi's class name
     * Return: a Kiwi's class name
     */
    public String getClassName() {
        return "Kiwi";
    }
   
    /* A Kiwi's Call
     * Return: a kiwi-call ("Chichi!").
     */
    public String call() {
        return "Chichi!";
    }
   
   }
   
   